R/check_result.R

```diff
@@ -30,23 +31,24 @@ check_result <- function(
   learnr_args = list() # provided by `grade_learnr`
 ) {
   results <- list(...)
-  chkm8_item_class(results, "grader_result")
+  chkm8_item_class(results, "grader_condition")
   chkm8_single_character(correct)
   chkm8_single_character(incorrect)
 
   if (!any(vapply(results, `[[`, logical(1), "correct"))) {
     stop("At least one correct result must be provided")
   }
 
-  user_answer <- learnr_args$last_value
-
   # init final answer as not found
   final_result <- graded(correct = FALSE, NULL)
   found_match <- FALSE
+
   for (resu in results) {
-    if (identical(resu$x, user_answer)) {
-      final_result <- resu
+    evaluated_condi <- evaluate_condi(resu, grader_args, learnr_args)
+    if (! is.null(evaluated_condi)) {
       found_match <- TRUE
+      final_result <- list(correct = resu$correct,
```

Comment:
If the`evaluated_condi` is not `NULL`, it should be a `graded` result.

Use 

```r
graded(correct = evaluated_condi$correct, message = evaluated_condi$message)
```

Reply:
fixes #29